src/module/apps/compendium-browser/tabs/base.ts

```diff
@@ -298,21 +298,23 @@ export abstract class CompendiumBrowserTab {
         initial?: number;
         weight?: number;
     }): Partial<TableResultSource>[] {
-        return this.currentIndex.flatMap((e, i) => {
-            const data = fromUuidSync(e.uuid);
-            if (!data?.pack || !data._id || !("name" in data)) return [];
-            const rangeMinMax = initial + i + 1;
-            return {
-                text: data.name,
-                type: CONST.TABLE_RESULT_TYPES.COMPENDIUM,
-                collection: data.pack,
-                resultId: data._id,
-                img: e.img,
-                weight,
-                range: [rangeMinMax, rangeMinMax],
-                drawn: false,
-            };
-        });
+        return this.currentIndex
+            .map((e, i): Partial<TableResultSource> | null => {
+                const data = fromUuidSync(e.uuid);
+                if (!data?.pack || !data._id || !("name" in data)) return null;
+                const rangeMinMax = initial + i + 1;
+                return {
+                    text: data.name,
+                    type: CONST.TABLE_RESULT_TYPES.COMPENDIUM,
+                    documentCollection: data.pack,
+                    documentId: data._id,
+                    img: e.img,
+                    weight,
+                    range: [rangeMinMax, rangeMinMax],
+                    drawn: false,
+                };
+            })
+            .filter((r): r is Partial<TableResultSource> => !!r);
```

Comment:
I was going to suggest a flatMap but then turned my head left :upside_down_face:

Reply:
for whatever reason, typescript just let flatMap() get the types completely wrong, which is how this bug even slipped through. Best I could do is keep the flat map and assign to a variable that has a specific type and return it.